src/widgets/attendance-options-select/ui/attendance-option-select.tsx

```diff
@@ -0,0 +1,38 @@
+'use client'
+
+import { Select } from '@mantine/core'
+import type { SelectProps } from '@mantine/core'
+import type { GetAttendanceOptionsAll } from '@/entities/attendance-option'
+
+export interface AttendanceOptionSelectProps extends SelectProps {
+  attendanceOptions: GetAttendanceOptionsAll
+}
+
+export const AttendanceOptionSelect: React.FC<AttendanceOptionSelectProps> = ({
+  attendanceOptions,
+  value,
+  onChange,
+  className,
+  classNames,
+  checkIconPosition = 'right',
+  allowDeselect = false,
+  ...props
+}) => {
+  const options = Object.entries(attendanceOptions).flatMap(([key, { name, detailOptions }]) => [
+    { value: key, label: name },
+    ...detailOptions.map(({ id, name }) => ({ value: String(id), label: name })),
+  ])
+
+  return (
+    <Select
+      data={options}
+      value={value}
+      onChange={onChange}
+      className={className}
+      classNames={classNames}
+      checkIconPosition={checkIconPosition}
+      allowDeselect={allowDeselect}
+      {...props}
+    />
+  )
+}
```

Comment:
[질문]
해당 props(onChange, className, classNames)는 아래와 같이 구조 분해 할당 문법을 사용하지 않아도 무방한 것 같다고 생각되는데 따로 표기하신 이유가 있으신가요?

<br />

```suggestion
export const AttendanceOptionSelect: React.FC<AttendanceOptionSelectProps> = ({
  attendanceOptions,
  value,
  checkIconPosition = 'right',
  allowDeselect = false,
  ...props
}) => {
  const options = Object.entries(attendanceOptions).flatMap(([key, { name, detailOptions }]) => [
    { value: key, label: name },
    ...detailOptions.map(({ id, name }) => ({ value: String(id), label: name })),
  ])

  return (
    <Select
      data={options}
      value={value}
      checkIconPosition={checkIconPosition}
      allowDeselect={allowDeselect}
      {...props}
    />
  )
}
```

Reply:
아니요, 딱히 이유는 없었고 기존에 form 내부에 작성했던 코드를 컴포넌트로 만들면서 props를 그대로 옮겼던 것 같습니다. 제안해주신 것처럼 꼭 표기하지 않아도 되는 props는 삭제하는 것이 좋을 것 같네요. 수정하겠습니다!